### ISSUE TITLE
Radio signal strength equal to 0

### ISSUE THREAD

**Issue by HamoRiddle** [microsoft/pxt-microbit#2471]
*Describe the bug*
Ask the microbit to shows the received radio signal strength and it returns 0.
*To reproduce*
I am trying to do the hot or cold game.
Here is the link of Hot or Cold: https://makecode.microbit.org/projects/hot-or-cold
I followed every step in the tutorial.(not the extended part)
*Additional information*
The microbit should detect the received packet signal strength and displace icon accoundingly. The microbit keep on displaying the icon as if the radio signal strength is the strongest(-42). When I change the program to ask it to send me the signal strength, it returns 0(the range should be -128 to -42).

Thank you.

**Comment by abchatra** [comment]
It shows zero to me as well. @pelikhan is this due to your recent radio changes? 

Small repro

https://makecode.microbit.org/_f3qbka8XFKMk
```JAVASCRIPT
input.onButtonPressed(Button.A, function () {
    radio.sendNumber(0)
})
radio.onReceivedNumber(function (receivedNumber) {
    basic.showNumber(radio.receivedPacket(RadioPacketProperty.SignalStrength))
})
radio.setGroup(10)
radio.setTransmitPower(7)
basic.forever(function () {
	
})
```

**Comment by FranklinWhale** [comment]
Before the changes in #2461, `PacketBuffer packet` in https://github.com/microsoft/pxt-microbit/blob/master/libs/radio/radio.cpp is set in `readRawPacket()`. After the changes, `packet` is never assigned. Unfortunately, `receivedSignalStrength()` depends on `packet`.

**Comment by abchatra** [comment]
fixed in 2.0.10.
